### PR TITLE
chore: remove support for hdfs

### PR DESCRIPTION
**Pull request by NicholasBlaskey**
## Description

Remove support for hdfs.

## Test Plan

No testing required as long as master + schemas build.

## Checklist

- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

File: docs/release-notes/remove-hdfs.rst [hunk 1]
```diff
@@ -0,0 +1,5 @@
+:orphan:
+
+**Breaking Change**
+
+-  HDFS checkpoint storage support has been removed in a future version.
```
Comment (rb-determined-ai) [inline docs/release-notes/remove-hdfs.rst:5]: this is not coherent I think

File: harness/setup.py [hunk 2]
```diff
@@ -28,7 +28,6 @@
         "certifi",
         "filelock",
         "google-cloud-storage",
-        "hdfs>=2.2.2",
```
Comment (rb-determined-ai) [inline harness/setup.py:31]: :tada:

**Review by rb-determined-ai — COMMENTED** [review]
Looks great but it's missing legacy config handling in pkg/schemas/expconf/legacy.go.

We're breaking hdfs experiments and we believe nobody will actually be affected.

But if we're wrong about that, and we don't have legacy config handled, then any user with any hdfs experiments laying around will have a broken webui forever until either we add legacy support or they purge those experiments from their database manually.

(because our master code parses experiment configs from the database automatically in so many queries).

Might I suggest that in the ParseLegacyConfig, that you generate a bogus sharedfs checkpoint storage instead?  That way things fail at runtime but the webui is unaffected.

File: docs/release-notes/remove-hdfs.rst (hunk 1 above)
Comment (NicholasBlaskey) [reply docs/release-notes/remove-hdfs.rst]: Yeah don't know what I was thinking...

**Review by rb-determined-ai — APPROVED** [review]
nice

**Pull request merged**